pkg/config/expand.go

```diff
@@ -93,6 +97,35 @@ func validateModuleInputs(mp modulePath, m Module, bp Blueprint) error {
 	return errs.OrNil()
 }
 
+func validateModulesAreUsed(bp Blueprint) error {
+	errs := Errors{}
+
+	used := map[ModuleID]bool{}
+	bp.WalkModules(func(m *Module) error {
+		for _, ref := range valueReferences(m.Settings.AsObject()) {
+			used[ref.Module] = true
+		}
+		return nil
+	})
+
+	for ig, g := range bp.DeploymentGroups {
+		for im, m := range g.Modules {
+			if !m.InfoOrDie().Metadata.Ghpc.HasToBeUsed {
+				continue
+			}
+
+			if !used[m.ID] {
+				errs.At(Root.Groups.At(ig).Modules.At(im).ID,
+					HintError{
+						"you need to add it to the `use`-block of downstream modules",
```

Comment:
what if I assign an output of this module to an input of the downstream module? Wouldn't that work too?

Reply:
Yes, that would work, module would be considered "used".  
 I intentionally keep the hint short to steer users toward `use`.